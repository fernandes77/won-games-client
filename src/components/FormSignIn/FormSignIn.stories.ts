import type { Meta, StoryObj } from '@storybook/react'
import { FormSignIn } from './FormSignIn'

const meta: Meta<typeof FormSignIn> = {
  component: FormSignIn
}

export default meta
type Story = StoryObj<typeof FormSignIn>

export const Primary: Story = {}
