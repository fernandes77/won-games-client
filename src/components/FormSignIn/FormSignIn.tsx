import { Button } from '@/components/Button/Button'
import { TextField } from '@/components/TextField/TextField'
import { IconLock, IconMail } from '@tabler/icons-react'
import Link from 'next/link'

export const FormSignIn = () => (
  <div>
    <form>
      <TextField
        name="email"
        placeholder="Email"
        type="email"
        icon={<IconMail />}
        className="my-xxs"
      />
      <TextField
        name="password"
        placeholder="Password"
        type="password"
        icon={<IconLock />}
        className="my-xxs"
      />
      <a
        href="#"
        className="block text-sm text-black text-right hover:text-gray-dark"
      >
        Forgot your password?
      </a>

      <Button className="mt-md mx-auto mb-xs" size="lg" fullWidth>
        Sign in now
      </Button>

      <div className="text-sm text-black text-center">
        Don't have an account?{' '}
        <Link
          className="text-secondary border-b-[1px] border-solid border-secondary
          transition-[color,border,0.1s_ease-in-out] hover:text-secondary-dark
          hover:border-b-[1px] hover:border-solid hover:border-secondary-dark"
          href="/sign-up"
        >
          Sign up
        </Link>
      </div>
    </form>
  </div>
)
