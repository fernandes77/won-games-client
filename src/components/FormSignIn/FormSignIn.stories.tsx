import type { Meta, StoryObj } from '@storybook/react'
import { FormSignIn } from './FormSignIn'

const meta: Meta<typeof FormSignIn> = {
  component: FormSignIn
}

export default meta
type Story = StoryObj<typeof FormSignIn>

export const Primary: Story = {
  parameters: { backgrounds: { default: 'won-light' } },
  decorators: [
    (Story) => (
      <div style={{ margin: '0 auto', maxWidth: '360px' }}>
        {/* ðŸ‘‡ Decorators in Storybook also accept a function. Replace <Story/> with Story() to enable it  */}
        <Story />
      </div>
    )
  ]
}
