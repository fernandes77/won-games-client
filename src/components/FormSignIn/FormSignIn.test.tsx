import { FormSignIn } from '@/components/FormSignIn/FormSignIn'
import { render, screen } from '@/utils/tests/helpers'

describe('<FormSignIn />', () => {
  it('should render the form', () => {
    const { container } = render(<FormSignIn />)

    expect(screen.getByPlaceholderText(/email/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /sign in now/i })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should render the forgot password link', () => {
    render(<FormSignIn />)

    expect(
      screen.getByRole('link', { name: /forgot your password\?/i })
    ).toBeInTheDocument()
  })

  it('should render text to sign up if already have an account', () => {
    render(<FormSignIn />)

    expect(
      screen.getByRole('link', { name: /create one/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/don't have an account\?/i)).toBeInTheDocument()
  })
})
