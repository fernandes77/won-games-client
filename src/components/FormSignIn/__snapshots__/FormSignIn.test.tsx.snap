// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render the form 1`] = `
<body>
  <div>
    <div>
      <form>
        <div
          class="my-xxs"
        >
          <div
            class="
        flex bg-gray-light rounded-sm px-xs border-2 border-solid border-gray-light
        focus-within:shadow-[0_0_0.5rem] focus-within:shadow-primary"
          >
            <div
              class="order-none
            
            
            flex items-center w-[1.375rem] text-gray [&>svg]:w-full"
            >
              <svg
                class="tabler-icon tabler-icon-mail"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 7a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v10a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-10z"
                />
                <path
                  d="M3 7l9 6l9 -6"
                />
              </svg>
            </div>
            <input
              class="pl-xs
          
           text-black font-poppins text-base py-xxs bg-transparent border-0 outline-none w-full"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="my-xxs"
        >
          <div
            class="
        flex bg-gray-light rounded-sm px-xs border-2 border-solid border-gray-light
        focus-within:shadow-[0_0_0.5rem] focus-within:shadow-primary"
          >
            <div
              class="order-none
            
            
            flex items-center w-[1.375rem] text-gray [&>svg]:w-full"
            >
              <svg
                class="tabler-icon tabler-icon-lock"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 13a2 2 0 0 1 2 -2h10a2 2 0 0 1 2 2v6a2 2 0 0 1 -2 2h-10a2 2 0 0 1 -2 -2v-6z"
                />
                <path
                  d="M11 16a1 1 0 1 0 2 0a1 1 0 0 0 -2 0"
                />
                <path
                  d="M8 11v-4a4 4 0 1 1 8 0v4"
                />
              </svg>
            </div>
            <input
              class="pl-xs
          
           text-black font-poppins text-base py-xxs bg-transparent border-0 outline-none w-full"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <a
          class="block text-sm text-black text-right hover:text-gray-dark"
          href="#"
        >
          Forgot your password?
        </a>
        <button
          class="inline-flex items-center justify-center cursor-pointer text-white rounded p-xxs [&>svg]:w-4 h-[3.125rem] text-md px-xl py-xxs w-full bg-gradient-to-t from-[#ff5f5f] via-[#f062c0] to-[#f062c0] hover:bg-gradient-to-t hover:from-[#e35565] hover:via-[#d958a6] hover:to-[#d958a6] mt-md mx-auto mb-xs"
        >
          <span
            class=""
          >
            Sign in now
          </span>
        </button>
        <div
          class="text-sm text-black text-center"
        >
          Don't have an account?
           
          <a
            class="text-secondary border-b border-solid border-secondary transition-[color,border,0.1s_ease-in-out] hover:text-secondary-dark hover:border-b hover:border-solid hover:border-secondary-dark"
            href="/sign-up"
          >
            Create one
          </a>
        </div>
      </form>
    </div>
  </div>
</body>
`;
